Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -121,15 +121,13 @@ def algebraMap (R : Type u) (A : Type v) [CommSemiring R] [Semiring A] [Algebra
   Algebra.toRingHom
 #align algebra_map algebraMap
 
-/-- Coercion from a commutative semiring to an algebra over this semiring. -/
-@[coe] def Algebra.cast {R A : Type*} [CommSemiring R] [Semiring A] [Algebra R A] : R → A :=
-  algebraMap R A
+attribute [coe] algebraMap
 
 namespace algebraMap
 
 scoped instance coeHTCT (R A : Type*) [CommSemiring R] [Semiring A] [Algebra R A] :
     CoeHTCT R A :=
-  ⟨Algebra.cast⟩
+  ⟨algebraMap _ _⟩
```

Comment:
The errors are cause by `norm_cast` not understanding `algebraMap`: according to `norm_cast` the coercion should be on the head symbol, which in `↑(algebraMap R A) x` is actually `coeFun`. So as long as we want this instance, we should keep a `def Algebra.cast`.

Reply:
I definitely want this instance! Thanks for the explanation!